render/render.c

```diff
@@ -471,18 +461,23 @@ ProcRenderQueryPictFormats(ClientPtr client)
         ++pictSubpixel;
     }
 
+    xRenderQueryPictFormatsReply reply = {
+        .numFormats = nformat,
+        .numScreens = numScreens,
+        .numDepths = ndepth,
+        .numVisuals = nvisual,
+        .numSubpixel = numSubpixel,
+    };
```

Comment:
Be careful using `ndepth` and `nvisual` here. Those variables are reused in the loops above and can change their value.

Reply:
oh, great finding, thanks.

I've prepended some commits for moving them out of the way.